src/wallet/spend.cpp

```diff
@@ -961,7 +961,15 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
             // If an input as a preset sequence, we can't do anti-fee-sniping
             use_anti_fee_sniping = false;
         }
-        txNew.vin.push_back(CTxIn(coin.outpoint, CScript(), sequence));
+        CScript script_sig;
+        CScriptWitness script_witness;
+        if (coin_control.HasScripts(coin.outpoint)) {
+            auto [ss, sw] = coin_control.GetScripts(coin.outpoint);
+            script_sig = ss;
+            script_witness = sw;
+        }
+        txNew.vin.push_back(CTxIn(coin.outpoint, script_sig, sequence));
+        txNew.vin.back().scriptWitness = script_witness;
```

Comment:
Right now, I don't think that's a good idea. Since we are always say that the preset inputs have scriptSigs and witnesses when they come from `FundTransaction`, any input weight calculation using them may be incorrect because those inputs are not necessarily complete or valid, so the final weight may actually differ.